docs/product/specs/permissions-revamp.md

```diff
@@ -0,0 +1,146 @@
+  - Admins should be able to update this user-to-role configuration.
+* All Mathesar admins should be able to access all databases configured in Mathesar.
+  - See 'Role for admins' section on how roles are used by Admins.
+* Admins cannot be removed from a database.
+* Standard Mathesar users can be added/removed from a database.
+* Standard Mathesar users will not be able to access a database without being provided explicit access by a Mathesar admin.
+
+#### Permission checks for objects
+* The permissions for DB objects will be determined by the priviliges of the role the user is associated with.
+* For Mathesar specific objects, we will have a setting per user per database that allows the user to 'read' or 'read-write'.
+  - This applies for explorations and metadata such as table and column settings.
+  - Future consideration:
+    - We can allow an additional option 'Infer from DB privileges' that will automatically identify access to each Mathesar-object based on the privileges on the underlying DB objects.
+
+#### Role for admins
+* An admin is not associated with a single role.
+* Admins should be able to utilize any role that's configured for a database.
+* When an admin opens a DB object, lets say a table,
+  - If the owning role of that table is configured, then any operation performed by the admin will utilize that role.
+  - If the owing role isn't configured, we'll try to figure out the best role for the admin to use. In cases where it's not possible, we will use the `default` role.
+* More information is provided in the 'Default role' and 'Dealing with Ownership and Collaboration issues' sections of this document.
+
+### Default role
+* There should be one default role per database.
+* When a database is first configured, the role used to connect to the database will be set as the default role.
+  - This role is used for installing/upgrading the functions on the DB.
+* The default role cannot be removed.
+  - This will ensure that the admins have atleast one role to connect to the database.
+* The default role cannot be changed. 
+  - We can provide this option in the future.
+  - This restriction is to reduce scope since changing the default role has more considerations such as changing ownerships for the installed functions, which might affect upgrades.
+
+### Dealing with Ownership and granting access
+#### Considerations
+* PostgreSQL has ownership for all DB objects.
+* Tables/schemas created by one role cannot be viewed/accessed by another role, unless the access is granted explicitly.
+* When tables/schemas are created via Mathesar UI, it may cause collaboration issues among Mathesar users if they utilize different roles.
+
+#### Approach for tables
+* When a table is created via the UI, the ownership will belong to the role of the Mathesar user that created that object.
+  * Admins will have access to all roles on the DB and hence will be able to use the role that created the table and obtain full ownership.
```

Comment:
I would say they're acting as the owner. They're not really obtaining anything, and we need to make sure they understand that in the UI. They'll get the wrong understanding if they think acting on the table somehow takes ownership from the user who created the table.